File: exemplos/js/02_eu.mjs
// Quem é o dono da chave e que escopos ela tem (GET /api/eu, escopo token:qualquer)
import { chamar, exigir, exigirChave } from "./_apoio.mjs";

exigirChave();
const { status, objeto } = await chamar("/api/eu");
exigir(status === 200, `esperava 200 em /api/eu, veio ${status}: ${JSON.stringify(objeto)}`);
exigir(objeto.token, "a resposta não traz o bloco token: a credencial usada foi sessão, não chave");
console.log(`chave ${JSON.stringify(objeto.token.nome)} do usuário ${objeto.login} no inquilino ${objeto.inquilino.slug}`);
console.log(`escopos: ${objeto.token.escopos.join(", ")}`);
console.log(`expira em: ${objeto.token.expira_em}`);

File: exemplos/python/02_eu.py
# Quem é o dono da chave e que escopos ela tem (GET /api/eu, escopo token:qualquer)
from _apoio import chamar, exigir, exigir_chave

exigir_chave()
status, _cab, corpo = chamar("/api/eu")
exigir(status == 200, f"esperava 200 em /api/eu, veio {status}: {corpo}")
exigir(corpo.get("token"), "a resposta não traz o bloco token: a credencial usada foi sessão, não chave")
escopos = corpo["token"]["escopos"]
print(f"chave {corpo['token']['nome']!r} do usuário {corpo['login']} no inquilino {corpo['inquilino']['slug']}")
print(f"escopos: {', '.join(escopos)}")
print(f"expira em: {corpo['token']['expira_em']}")
